src/Sync.ts

```diff
@@ -0,0 +1,200 @@
+    this._synchronizedFlickings = val;
+    this._addEvents(this._synchronizedFlickings);
+  }
+
+  public constructor({
+    type = SYNC.TYPE.CAMERA,
+    synchronizedFlickings = [],
+  }: Partial<SyncOptions> = {}) {
+    this._type = type;
+    this._synchronizedFlickings = synchronizedFlickings;
+  }
+
+  public init(flicking: Flicking): void {
+    if (this._flicking) {
+      this.destroy();
+    }
+
+    this._flicking = flicking;
+
+    this._addEvents(this._synchronizedFlickings);
+
+    this._synchronizedFlickings.forEach(synchronizedFlicking => {
+      this._updateClass(synchronizedFlicking, 0);
+    });
+  }
+
+  public destroy(): void {
+    const flicking = this._flicking;
+
+    if (!flicking) {
+      return;
+    }
+
+    this._removeEvents(this._synchronizedFlickings);
+
+    this._flicking = null;
+  }
+
+  public update(): void {
+    this._synchronizedFlickings.forEach(synchronizedFlicking => {
+      this._updateClass(synchronizedFlicking, synchronizedFlicking.flicking.camera.position);
+    });
+  }
+
+  private _addEvents = (synchronizedFlickings: SychronisableFlickingOptions[]) => {
+    synchronizedFlickings.forEach((synchronizedFlicking) => {
+      if (this._type === "camera") {
+        synchronizedFlicking.flicking.on(EVENTS.MOVE, this._onMove);
+        synchronizedFlicking.flicking.on(EVENTS.MOVE_START, this._onMoveStart);
+        synchronizedFlicking.flicking.on(EVENTS.MOVE_END, this._onMoveEnd);
+      }
+      if (this._type === "index" && synchronizedFlicking.isSlidable) {
+        synchronizedFlicking.flicking.on(EVENTS.WILL_CHANGE, this._onChangeIndex);
+      }
+      if (this._type === "index" && synchronizedFlicking.isClickable) {
+        synchronizedFlicking.flicking.on(EVENTS.SELECT, this._onChangeIndex);
+      }
+    });
+  };
+
+  private _removeEvents = (synchronizedFlickings: SychronisableFlickingOptions[]) => {
+    synchronizedFlickings.forEach((synchronizedFlicking) => {
+      if (this._type === "camera") {
+        synchronizedFlicking.flicking.off(EVENTS.MOVE, this._onMove);
+        synchronizedFlicking.flicking.off(EVENTS.MOVE_START, this._onMoveStart);
+        synchronizedFlicking.flicking.off(EVENTS.MOVE_END, this._onMoveEnd);
+      }
+      if (this._type === "index" && synchronizedFlicking.isSlidable) {
+        synchronizedFlicking.flicking.off(EVENTS.WILL_CHANGE, this._onChangeIndex);
+      }
+      if (this._type === "index" && synchronizedFlicking.isClickable) {
+        synchronizedFlicking.flicking.off(EVENTS.SELECT, this._onChangeIndex);
+      }
+    });
+  };
+
+  private _onChangeIndex = (e: ComponentEvent) => {
+    const flicking = e.currentTarget;
+    const synchronizedFlickings = this._synchronizedFlickings;
+
+    if (!flicking.initialized || synchronizedFlickings.reduce((isAnimationComplete, synchronizedFlicking) => isAnimationComplete || (synchronizedFlicking.flicking !== flicking && synchronizedFlicking.flicking.animating), false)) {
```

Comment:
너무 긴 구문인데 적절한 이름을 만드는건 어떤가요?
It's a long phrase, but how about a proper name?

Reply:
animating을 판별하는 것 / 하나라도 해당하는지 체크하는 것 / flicking이 다른것만 검사하는 것
이렇게 목적별로 분리한 다음 조합하는건 어떨까요?
Determining animating / Checking if any one is true / Filtering only other flicking
How about separating them by purpose and then combining them?

```ts
const any =
  <T>(fn: (item: T) => boolean) =>
  (arr: T[]) =>
    arr.reduce((result, item) => result || fn(item), false);

const isAnimating = synchronizedFlicking => synchronizedFlicking.flicking.animating;
const isAnyAnimating = any(isAnimating);
const isAnySyncedAnimating = isAnyAnimating(synchronizedFlickings.filter(x => x.flicking !== flicking))
```

Reply:
현재는 https://github.com/naver/egjs-flicking-plugins/pull/18#discussion_r668366963 코멘트의 내용을 적용하는 과정에서 애니메이션 재생 여부를 판단하는 구문을 사용하지 않게 되었습니다.
말씀을 남겨주신 내용들은 이후 애니메이션 적용 여부를 다시 판단해야 하는 경우 참고할 수 있도록 하겠습니다.

Currently,  The code for checking animations is no longer used in the process of applying the comment https://github.com/naver/egjs-flicking-plugins/pull/18#discussion_r668366963.
These comments will be used as a reference in case we need to decide whether to apply the animation again in the future.